db/query.go

```diff
@@ -204,9 +201,11 @@ func (context *DatabaseContext) N1QLQueryWithStats(queryName string, statement s
 
 	results, err = gocbBucket.Query(statement, params, consistency, adhoc)
 	if err != nil {
-		queryExpvars.Add(fmt.Sprintf(n1qlQueryErrorCountExpvarFormat, queryName), 1)
+		context.DbStats.StatsGsiViews().Add(fmt.Sprintf(n1qlQueryErrorCountExpvarFormat, queryName), 1)
 	}
-	queryExpvars.Add(fmt.Sprintf(n1qlQueryCountExpvarFormat, queryName), 1)
+
+	context.DbStats.StatsGsiViews().Add(fmt.Sprintf(n1qlQueryCountExpvarFormat, queryName), 1)
```

Comment:
Originally I thought this should be removed (since we're adding a count by query type for channels/access/roleAccess/housekeeping).  But I'm thinking now we could leave it in, and treat 'housekeeping' as a calculated stat that's (total - (channels+access+roleAccess)).  If you agree I'll update the tracking doc.